components/shared/BackButton.tsx

```diff
@@ -0,0 +1,29 @@
+import { useRouter } from "next/router";
+import { FC } from "react";
+
+interface Props {
+    returnTo: string;
+}
+
+export const BackButton: FC<Props> = ({ returnTo }) => {
+
+    const router = useRouter();
+
+    const navigateBack = () => {
+        router.push(returnTo);
+    }
+
+    return (
+        <button
+            type="button"
+            className="inline-flex items-center rounded-full border border-transparent p-2 hover:bg-gray-100 focus:outline-none focus:ring-offset-2"
+            onClick={ navigateBack }
```

Comment:
quick recommendation: you'll want to get used to defining event handlers inline instead of declaring it above in a separate variable. It could be considered a matter of preference but given how Typescript works in general you'll want to specify as few things as possible. In this case if you put it inline you'll only be saving yourself having to name the handler (navigateBack) but in other cases by defining it inline you'll take advantage of Typescript and it will implicitly get the argument types without you having to define anything, and that's 1000% valuable